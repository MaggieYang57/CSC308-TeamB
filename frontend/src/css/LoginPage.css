@import url("https://fonts.googleapis.com/css2?family=Quicksand&display=swap");

.auth-form {
  display: flex;
  align-items: center;
  width: 700px;
  margin: 100px calc(50vw - 350px) 50px;
  padding: 0px 10px 40px;
  flex-direction: column;

  background-color: white;
  background-clip: padding-box;
  filter: drop-shadow(5px 0px 10px grey);
  overflow: hidden;
}

.title {
  padding-top: 50px;
}
.title h1 {
  letter-spacing: 2px;
  font-weight: 700;
  font-size: 40px;
}

p {
  font-family: "Quicksand", sans-serif;
  font-style: normal;
  font-size: 17px;
  color: black;
}

#cta-type {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-bottom: 20px;
  align-self: center;
}
#cta-type label {
  margin-left: 10px;
  margin-right: 20px;
  font-family: "Quicksand", sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  letter-spacing: 0.05em;
  color: #7e7c7c;
}

.input-email {
  justify-content: left;
  text-align: left;
  padding-right: 440px;
  margin-bottom: 10px;
  font-family: "Quicksand", sans-serif;
  font-weight: 700;
  font-size: 20px;
  letter-spacing: 2px;
  color: #7e7c7c;
}
.input-password {
  justify-content: left;
  text-align: left;
  padding-right: 395px;
  font-family: "Quicksand", sans-serif;
  font-weight: 700;
  font-size: 20px;
  letter-spacing: 2px;
  color: #7e7c7c;
}
input {
  padding: 12px 20px;
  margin: 8px;
  border: none;
}
input[type="text"],
input[type="password"],
input[type="email"],
input[type="tel"] {
  font-family: "Quicksand", sans-serif;
  font-size: 20px;
  background-color: #f5f5f5;
}
.login-form input[type="text"]:focus,
.login-form input[type="email"]:focus,
.login-form input[type="password"]:focus {
  border-width: 0px;
  border: none;
  background-color: #d3d3d3;
}
input[type="checkbox"] {
  -webkit-transform: scale(2);
  transform: scale(2);
}
input[type="radio"] {
  -webkit-transform: scale(1.5); /* Chrome, Safari, Opera */
  transform: scale(1.5);
  margin-right: 15px;
}

.link {
  display: inline;
  margin-top: -32px;
  margin-bottom: 10px;
  padding-left: 370px;
  font-family: "Quicksand", sans-serif;
  font-style: normal;
  font-weight: 800;
  font-size: 15px;
  color: black;
}

.password-security {
  margin: -10px;
  font-family: "Quicksand", sans-serif;
  font-weight: 400;
  font-size: 20px;
  color: #7e7c7c;
  display: block;
}
#password-visibility {
  margin-right: 20px;
}

#login-button {
  display: block;
  margin-bottom: 30px;
  overflow: hidden;
  padding: 12px 210px;
  border: none;
  font-weight: 600;
  font-size: 20px;
  letter-spacing: 2px;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: color, background-color;
  transition-property: color, background-color;
}
#login-button:hover,
#signin button-fade:focus,
#signin button-fade:active {
  background-color: #2098d1;
  color: white;
}

.error {
  margin: -10px 0px 20px;
  font-family: "Quicksand", sans-serif;
  font-style: normal;
  font-weight: 800;
  font-size: 17px;
  color: rgb(241, 80, 80);
}
