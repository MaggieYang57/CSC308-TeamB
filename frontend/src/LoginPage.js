/* eslint-disable node/handle-callback-err */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link, withRouter } from "react-router-dom";
import "./css/LoginPage.css";

require('dotenv').config()
const backendHostURL = process.env.REACT_APP_BACKEND_HOST_URL

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: false,
      RedirectLoggedUser: false,
      userType: "",
      emptyUser: false,
      email: "",
      password: "",
      error: false,
    };
  }

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  };

  isNotLoggedIn = () => {
    this.setState({ isLoggedIn: false });
  };

  isLoggedIn = () => {
    this.setState({ isLoggedIn: true });
  };

  passwordVisibility = () => {
    const password = document.getElementById("password");
    password.type =
      password.type === "password"
        ? (password.type = "text")
        : (password.type = "password");
  };

  changeUserType = (event) => {
    this.setState({ userType: event.target.value, emptyUser: false });
  };

  componentDidMount() {
    const path = window.location.pathname.split("/");

    if (path.length === 3) {
      if (path[2] === "admin") {
        document.getElementById("admin").checked = true;
        this.setState({ userType: path[2] });
      } else if (path[2] === "user") {
        document.getElementById("user").checked = true;
        this.setState({ userType: path[2] });
      }
    }
  }

  login = (e) => {
    e.preventDefault();

    if (this.state.userType === "") {
      this.setState({ emptyUser: true });
      return;
    }

    const user = {
      email: this.state.email,
      password: this.state.password,
      userType: this.state.userType,
    };

    this.mongoLogin(user);
  };

  mongoLogin = (user) => {
    const _this = this;
    fetch(`${backendHostURL}/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(user),
    })
      .then((res) => {
        if (res.status === 404) {
          _this.setState({ error: true });
        } else {
          return res.json();
        }
      })
      .then((data) => {
        _this.storeUser(data);
        this.props.history.push("/profile");
      })
      .catch((err) => {
        console.log("Error");
        _this.setState({ error: true });
      });
  };

  storeUser(user) {
    localStorage.setItem("email", user.user_email);
    localStorage.setItem("userType", JSON.stringify(user.userType));
    localStorage.setItem("isLoggedIn", "true");
    localStorage.setItem("_id", user._id);
    this.props.onUserChange(user.userType);
  };

  render() {
    return (
      <form className="auth-form" onSubmit={this.login}>
        <div className="title">
          {window.location.pathname.split("/")[2] === "admin" ? "Admin " : ""}
          {window.location.pathname.split("/")[2] === "user" ? "User " : ""}
          <b
            className="text text-center "
            style={{ fontSize: 50, color: "#2C6674" }}
          >
            Log In
          </b>
        </div>
        <div id="cta-type">
          <div id="admin">
            <input
              type="radio"
              id="admin"
              name="cta"
              value="admin"
              onChange={this.changeUserType}
              checked={null}
            />
            <label htmlFor="admin">Admin</label>
          </div>
          <div id="user">
            <input
              type="radio"
              id="user"
              name="cta"
              value="user"
              onChange={this.changeUserType}
              checked={null}
            />
            <label htmlFor="user">User</label>
          </div>
        </div>
        <p className="input-email">Email</p>
        <input
          type="text"
          id="email"
          size="50"
          style={{ width: "500px" }}
          onChange={this.handleChange}
        />
        <br />
        <p className="input-password">Password</p>
        <div className="link">
          <Link to="/reset-password">Forgot Password?</Link>
        </div>
        <input
          type="password"
          id="password"
          size="50"
          style={{ width: "500px" }}
          onChange={this.handleChange}
        />
        <br />
        <label className="password-security">
          <input
            type="checkbox"
            id="password-visibility"
            onClick={() => this.passwordVisibility()}
          />
          Show Password
        </label>
        <br />
        {this.state.emptyUser && (
          <div className="error">Select the type of user</div>
        )}
        {this.state.error && (
          <div className="error">Invalid email or password</div>
        )}
        <button id="login-button" type="submit">
          LOG IN
        </button>
        <p>
          Don&apos;t have an account? <Link to="/signup">Sign up</Link>
        </p>
      </form>
    );
  }
}

Login.propTypes = {
  history: PropTypes.object,
  userType: PropTypes.object,
  onUserChange: PropTypes.func,
};
export default withRouter(Login);
