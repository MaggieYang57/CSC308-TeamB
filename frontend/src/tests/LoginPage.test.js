import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import LoginPage from '../LoginPage';

it('renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter> <LoginPage /> </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});