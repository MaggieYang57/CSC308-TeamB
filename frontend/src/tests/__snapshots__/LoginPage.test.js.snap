// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <form
    className="auth-form"
    onSubmit={[Function]}
  >
    <div
      className="title"
    >
      
      
      <b
        className="text text-center "
        style={
          Object {
            "color": "#2C6674",
            "fontSize": 50,
          }
        }
      >
        Log In
      </b>
    </div>
    <div
      id="cta-type"
    >
      <div
        id="admin"
      >
        <input
          checked={null}
          id="admin"
          name="cta"
          onChange={[Function]}
          type="radio"
          value="admin"
        />
        <label
          htmlFor="admin"
        >
          Admin
        </label>
      </div>
      <div
        id="user"
      >
        <input
          checked={null}
          id="user"
          name="cta"
          onChange={[Function]}
          type="radio"
          value="user"
        />
        <label
          htmlFor="user"
        >
          User
        </label>
      </div>
    </div>
    <p
      className="input-email"
    >
      Email
    </p>
    <input
      id="email"
      onChange={[Function]}
      size="50"
      style={
        Object {
          "width": "500px",
        }
      }
      type="text"
    />
    <br />
    <p
      className="input-password"
    >
      Password
    </p>
    <input
      id="password"
      onChange={[Function]}
      size="50"
      style={
        Object {
          "width": "500px",
        }
      }
      type="password"
    />
    <br />
    <label
      className="password-security"
    >
      <input
        id="password-visibility"
        onClick={[Function]}
        type="checkbox"
      />
      Show Password
    </label>
    <br />
    <button
      id="login-button"
      type="submit"
    >
      LOG IN
    </button>
    <p>
      Don't have an account? 
      <a
        href="/signup"
        onClick={[Function]}
      >
        Sign up
      </a>
    </p>
  </form>,
  " ",
]
`;
